### PR TITLE
Setup modular cc toolchains repo

### PR DESCRIPTION
**Pull request by nathaniel-brough**
This doesn't actually implement any of the additions mentioned in the proposal just the skeleton structure to add to.

### CODE REVIEW

**Comment by nathaniel-brough** [comment]
Just got to get this [workflow](https://github.com/silvergasp/modular_cc_toolchains/actions/runs/3963973424) and then I'll add a CI branch protection rule.

**Comment by nathaniel-brough** [comment]
The CI is now passing. It won't show up here until the PR is merged.

https://github.com/silvergasp/modular_cc_toolchains/actions/runs/3964001103/jobs/6792460389

File: README.md [hunk 1]
```diff
@@ -1 +1,24 @@
...
+
+**NOTE**: Expect breaking change.
+
+## Goals
+A good rundown of why this repository exists is described in the 
+[modular cc toolchains proposal](https://docs.google.com/document/d/1-etGNsPneQ8W7MBMxtLloEq-Jj9ng1G-Pip-cWtTg_Y/edit?usp=sharing).
+
+The end goal will be to merge this into the 
+[rules_cc repository](https://github.com/bazelbuild/rules_cc). We think that we
+can implement the suggested changes with needing to make any breaking changes.
```
Comment (tpudlik) [inline README.md]: *without* needing to make breaking changes!

**Review by tpudlik — APPROVED** [review]
Sorry about the slow review, I had my notifications set up incorrectly and missed the email about this PR!

File: README.md (hunk 1 above)
Comment (nathaniel-brough) [reply README.md]: Oops hahaha

**Comment by nathaniel-brough** [comment]
No worries. I've had  few other things to work on anyway :)

**Review by tpudlik — APPROVED** [review]

**Pull request merged**